### DISCUSSION TITLE
How to run steps after a failing step

### DISCUSSION THREAD

**Discussion by duc-gp** [community/community#26739] [Actions]
<p>Scenario:</p>
<p>I have a simple workflow which should do the following:</p>
<ul>
<li>
<p>checkout</p>
</li>
<li>
<p>install cypress</p>
</li>
<li>
<p>run cypress</p>
</li>
<li>
<p>upload screenshots done by cypress</p>
</li>
</ul>
<p>My problem is, that the last step of uploading is not done because the cypress failed due to failing tests</p>
<p>is it possible and if, how to run the steps even if previous steps failed? </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9f99705339768e9016298cfabf2d7d59756cb983.png" data-download-href="/uploads/short-url/mLSEQClL5q4wsm9WphkLtEVHMQP.png?dl=1" title="Screenshot 2019-09-20 at 14.15.59.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/40763918/181099674-f036d4b0-3e1a-47d6-8068-c3df7a5460f6.png" alt="Screenshot 2019-09-20 at 14.15.59.png" data-base62-sha1="mLSEQClL5q4wsm9WphkLtEVHMQP" width="690" height="133" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9f99705339768e9016298cfabf2d7d59756cb983_2_690x133.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/9f99705339768e9016298cfabf2d7d59756cb983_2_1035x199.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/9f99705339768e9016298cfabf2d7d59756cb983.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9f99705339768e9016298cfabf2d7d59756cb983_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2019-09-20 at 14.15.59.png</span><span class="informations">1103×214 9.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by bendavies** [comment, marked as answer]
<p>here you go:</p>
<p><a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#job-status-check-functions" rel="nofollow noopener">https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#job-status-check-functions</a></p>

**Comment by gqqnbig** [comment]
<p>Can I assume by default all steps have an implicit </p>
<p>if: ${{ success() }}</p>

**Comment by negebauer** [comment]
<p>Yes, <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions" rel="nofollow noopener">https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions</a></p>
<blockquote>
<p>You can use the following status check functions as expressions in  <code>if</code>  conditionals. If your  <code>if</code>  expression does not contain any of the status functions it will automatically result with  <code>success()</code> . For more information about  <code>if</code>  conditionals, see “<a href="https://docs.github.com/en/articles/workflow-syntax-for-github-actions/#jobsjob_idif" rel="nofollow noopener">Workflow syntax for GitHub Actions</a>.”</p>
</blockquote>

**Comment by danielcompton** [comment]
<aside class="quote no-group" data-username="negebauer" data-post="4" data-topic="16309">
<div class="title">
<div class="quote-controls"></div>
 negebauer:</div>
<blockquote>
<p>You can use the following status check functions as expressions in <code>if</code> conditionals. If your <code>if</code> expression does not contain any of the status functions it will automatically result with <code>success()</code> . For more information about <code>if</code> conditionals, see “<a href="https://docs.github.com/en/articles/workflow-syntax-for-github-actions/#jobsjob_idif" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions </a>.”</p>
</blockquote>
</aside>
<p>For future searches, those docs have been moved to <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#job-status-check-functions" class="inline-onebox" rel="noopener nofollow ugc">Expressions - GitHub Docs</a>.</p>
